darwin/future/data_objects/workflow.py

```diff
@@ -0,0 +1,172 @@
+from datetime import datetime
+from enum import Enum
+from typing import List, Optional
+from uuid import UUID
+
+from pydantic import Field
+
+from darwin.future.data_objects.typing import UnknownType
+from darwin.future.pydantic_base import DefaultDarwin
+
+
+class WFDataset(DefaultDarwin):
+    """
+    A class to manage all the information around a dataset on the darwin platform,
+    including validation
+
+    Attributes
+    ----------
+    id: int
+    name: str
+    instructions: str
+
+    Methods
+    ----------
+    __int__: returns the id of the dataset
+    __str__: returns the name of the dataset
+    __repr__: returns a string representation of the dataset
+    """
+
+    id: int
+    name: str
+    instructions: str = Field(min_length=0)
+
+    def __int__(self) -> int:
+        return self.id
+
+    def __str__(self) -> str:
+        return self.name
+
+
+class WFEdge(DefaultDarwin):
+    """
+    A workflow edge
+
+    Attributes
+    ----------
+
+    id: str
+    name: str
+    source_stage_id: str
+    target_stage_id: str
+    """
+
+    id: UUID
+    name: str
+    source_stage_id: UUID
+    target_stage_id: UUID
+
+
+class WFType(Enum):
+    """
+    The type of workflow stage (Enum)
+
+    Enumerations
+    ----------
+    DATASET: str
+    ANNOTATE: str
+    REVIEW: str
+    COMPLETE: str
+    """
+
+    # TODO: There may be more types
+    DATASET = "dataset"
+    ANNOTATE = "annotate"
+    REVIEW = "review"
+    COMPLETE = "complete"
+
+
+class WFUser(DefaultDarwin):
+    stage_id: UUID
```

Comment:
This seems complete by the data structures in the JSON, which are concise and clear in this case